internal/pkg/telemetry/ddprovider.go

```diff
@@ -0,0 +1,31 @@
+package telemetry
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+type singleTracerProvider struct {
+	tracer trace.Tracer
+}
+
+// WrapDD is a workaround for DataDog OpenTelemetry tracer.
+// DataDog restarts a global tracer on each TracerProvider.Tracer() call, which is not what we want.
+// In the DataDog library there is no concept (internally yes, but not publicly) of tracer instance,
+// everything is handled globally.
+func WrapDD(tracer trace.Tracer) trace.TracerProvider {
+	return &singleTracerProvider{tracer: &wrappedDDTracer{tracer: tracer}}
+}
+
```

Comment:
DataDog pre Go ma tieto problemy:
- Je cely globalny, neda sa vytvorit instancia (aj ked interne tam je, neviem preco je to private).
- OTEL DD TracerProvider ma metodu `Tracer(name)`, ktora vrati vzdy ten isty globalny tracer.
- Problem je, ze pri kazdom volani `Tracer(name)` sa cely setup DD tracingu sa zresetuje, a `Tracer` sa vola u nas viac krat, vola ho kazda integracia, `go-client`, `otelhttp`, ... 
- Vid `tracer.Start` v DD kode: https://github.com/DataDog/dd-trace-go/blob/596353a907da3ce3f5c2e554e2ac252dffbf8913/ddtrace/opentelemetry/tracer_provider.go#L60-L71

Fix:
- Vytvoreny `singleTracerProvider`, ktory vrati vzdy ten isty DD tracer ale nerestartuje DD tracing.